### PR TITLE
Add optional nftEndpoint param

### PR DESCRIPTION
**Pull request by octave08**
### 📦 Pull Request

[Provide a general summary of the pull request here.]

Create a magic instance with `nftEnpoint` param so that internal developers point to nft-api endpoint they want 

### ✅ Fixed Issues

- [List any fixed issues here like: Fixes #XXXX]

Add `nftEndpoint` param in SDK config

### 🚨 Test instructions

[Describe any additional context required to test the PR/feature/bug fix.]

### ⚠️ Don't forget to add a [semver](https://semver.org/) label! 
Please only add **one** label:
- `patch`: Bug Fix?
- `minor`: New Feature?
- `major`: Breaking Change?
- `skip-release`: It's unnecessary to publish this change.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @magic-ext/algorand@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/aptos@1.5.0-canary.586.5627642781.0
  npm install @magic-ext/auth@1.5.0-canary.586.5627642781.0
  npm install @magic-ext/avalanche@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/bitcoin@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/conflux@11.5.0-canary.586.5627642781.0
  npm install @magic-ext/cosmos@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/ed25519@9.5.0-canary.586.5627642781.0
  npm install @magic-ext/flow@13.6.0-canary.586.5627642781.0
  npm install @magic-ext/gdkms@1.5.0-canary.586.5627642781.0
  npm install @magic-ext/harmony@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/icon@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/near@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/oauth@12.5.0-canary.586.5627642781.0
  npm install @magic-ext/oidc@1.4.0-canary.586.5627642781.0
  npm install @magic-ext/polkadot@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/react-native-bare-oauth@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/react-native-expo-oauth@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/solana@14.5.0-canary.586.5627642781.0
  npm install @magic-ext/taquito@10.5.0-canary.586.5627642781.0
  npm install @magic-ext/terra@10.5.0-canary.586.5627642781.0
  npm install @magic-ext/tezos@13.5.0-canary.586.5627642781.0
  npm install @magic-ext/webauthn@12.5.0-canary.586.5627642781.0
  npm install @magic-ext/zilliqa@13.5.0-canary.586.5627642781.0
  npm install @magic-sdk/commons@14.5.0-canary.586.5627642781.0
  npm install @magic-sdk/pnp@12.5.0-canary.586.5627642781.0
  npm install @magic-sdk/provider@18.5.0-canary.586.5627642781.0
  npm install @magic-sdk/react-native-bare@19.5.0-canary.586.5627642781.0
  npm install @magic-sdk/react-native-expo@19.5.0-canary.586.5627642781.0
  npm install @magic-sdk/types@15.7.0-canary.586.5627642781.0
  npm install magic-sdk@18.5.0-canary.586.5627642781.0
  # or 
  yarn add @magic-ext/algorand@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/aptos@1.5.0-canary.586.5627642781.0
  yarn add @magic-ext/auth@1.5.0-canary.586.5627642781.0
  yarn add @magic-ext/avalanche@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/bitcoin@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/conflux@11.5.0-canary.586.5627642781.0
  yarn add @magic-ext/cosmos@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/ed25519@9.5.0-canary.586.5627642781.0
  yarn add @magic-ext/flow@13.6.0-canary.586.5627642781.0
  yarn add @magic-ext/gdkms@1.5.0-canary.586.5627642781.0
  yarn add @magic-ext/harmony@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/icon@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/near@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/oauth@12.5.0-canary.586.5627642781.0
  yarn add @magic-ext/oidc@1.4.0-canary.586.5627642781.0
  yarn add @magic-ext/polkadot@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/react-native-bare-oauth@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/react-native-expo-oauth@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/solana@14.5.0-canary.586.5627642781.0
  yarn add @magic-ext/taquito@10.5.0-canary.586.5627642781.0
  yarn add @magic-ext/terra@10.5.0-canary.586.5627642781.0
  yarn add @magic-ext/tezos@13.5.0-canary.586.5627642781.0
  yarn add @magic-ext/webauthn@12.5.0-canary.586.5627642781.0
  yarn add @magic-ext/zilliqa@13.5.0-canary.586.5627642781.0
  yarn add @magic-sdk/commons@14.5.0-canary.586.5627642781.0
  yarn add @magic-sdk/pnp@12.5.0-canary.586.5627642781.0
  yarn add @magic-sdk/provider@18.5.0-canary.586.5627642781.0
  yarn add @magic-sdk/react-native-bare@19.5.0-canary.586.5627642781.0
  yarn add @magic-sdk/react-native-expo@19.5.0-canary.586.5627642781.0
  yarn add @magic-sdk/types@15.7.0-canary.586.5627642781.0
  yarn add magic-sdk@18.5.0-canary.586.5627642781.0
  ```
</details>

### CODE REVIEW

**Review by karljin — APPROVED** [review]

File: packages/@magic-sdk/provider/test/spec/modules/user/getInfo.spec.ts [hunk 1]
```diff
@@ -21,3 +21,16 @@ test('Generate JSON RPC request payload with method `magic_get_info` and the act
...
+  magic.user.request = jest.fn();
+
+  await storage.setItem('mc_active_wallet', 'metamask');
+
+  await magic.user.getInfo();
+
+  const requestPayload = magic.user.request.mock.calls[0][0];
+  expect(requestPayload.method).toBe('magic_get_info');
+  expect(requestPayload.params).toEqual([{ walletType: 'metamask' }]);
+});
```
Comment (octave08) [inline packages/@magic-sdk/provider/test/spec/modules/user/getInfo.spec.ts:36]: This code is not related to this PR, but it helps this PR to pass the coverage threshold.

**Review by bengriffin1 — DISMISSED** [review]

**Comment by octave08** [comment]
@Ariflo thanks for letting me know 
I rebased this PR on the changed merge-base

btw
Do you know how to solve the Canary build failure?

**Comment by Ariflo** [comment]
> Do you know how to solve the Canary build failure?

@octave08 we believe its currently failing due to a recent change to our CI authentication token. Infra is currently in process of getting it fixed. Hoping this get resolved today, if not early tomorrow. 

CC: @darrentmagic @Ethella

**Review by bengriffin1 — APPROVED** [review]

**Review by karljin — APPROVED** [review]

**Comment by Ariflo** [comment]
:rocket: PR was released in [`@magic-ext/algorand@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/algorand%4013.5.0), [`@magic-ext/aptos@1.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/aptos%401.5.0), [`@magic-ext/auth@1.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/auth%401.5.0), [`@magic-ext/avalanche@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/avalanche%4013.5.0), [`@magic-ext/bitcoin@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/bitcoin%4013.5.0), [`@magic-ext/conflux@11.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/conflux%4011.5.0), [`@magic-ext/cosmos@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/cosmos%4013.5.0), [`@magic-ext/ed25519@9.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/ed25519%409.5.0), [`@magic-ext/flow@13.6.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/flow%4013.6.0), [`@magic-ext/gdkms@1.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/gdkms%401.5.0), [`@magic-ext/harmony@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/harmony%4013.5.0), [`@magic-ext/icon@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/icon%4013.5.0), [`@magic-ext/near@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/near%4013.5.0), [`@magic-ext/oauth@12.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/oauth%4012.5.0), [`@magic-ext/oidc@1.4.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/oidc%401.4.0), [`@magic-ext/polkadot@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/polkadot%4013.5.0), [`@magic-ext/react-native-bare-oauth@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/react-native-bare-oauth%4013.5.0), [`@magic-ext/react-native-expo-oauth@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/react-native-expo-oauth%4013.5.0), [`@magic-ext/solana@14.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/solana%4014.5.0), [`@magic-ext/taquito@10.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/taquito%4010.5.0), [`@magic-ext/terra@10.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/terra%4010.5.0), [`@magic-ext/tezos@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/tezos%4013.5.0), [`@magic-ext/webauthn@12.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/webauthn%4012.5.0), [`@magic-ext/zilliqa@13.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/zilliqa%4013.5.0), [`@magic-sdk/commons@14.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/commons%4014.5.0), [`@magic-sdk/pnp@12.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/pnp%4012.5.0), [`@magic-sdk/provider@18.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%4018.5.0), [`@magic-sdk/react-native-bare@19.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native-bare%4019.5.0), [`@magic-sdk/react-native-expo@19.5.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native-expo%4019.5.0), [`@magic-sdk/types@15.7.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%4015.7.0), [`magic-sdk@18.5.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%4018.5.0) :rocket:

**Pull request merged**